vizgrimoire/analysis/events_punchcard.py

```diff
@@ -102,18 +105,18 @@ def result(self):
             current_group = group
             if current_group <> old_group:
                 # we need to restart the process
-                punchcard[old_group] = week_structure.copy()
+                punchcard[old_group] = deepcopy(week_structure)
                 # Init data structure
-                week_structure = empty_week_structure.copy()
+                week_structure = deepcopy(empty_week_structure)
 
             weekday = int(data["weekday"][count])
             hour = int(data["hour"][count])
             mean = float(data["mean"][count])
             week_structure[weekday][hour] = mean
             count = count + 1
-            if count == len(data["group_name"]) -1:
```

Comment:
We were missing a Meetup group (the last one). Besides that imagine we have a list of 1 element, with "-1" the condition never happens.